src/coreComponents/dataRepository/xmlWrapper.cpp

```diff
@@ -169,6 +194,265 @@ string buildMultipleInputXML( string_array const & inputFileList,
+  size_t offsetInLine = npos;
+  size_t offset = npos;
+  xmlAttribute filePathAtt = node.attribute( filePathString );
+  xmlAttribute charOffsetAtt = node.attribute( charOffsetString );
+  string filePath;
+
+  if( filePathAtt && charOffsetAtt )
+  {
+    filePath = string( filePathAtt.value() );
+    offset = std::atoi( charOffsetAtt.value() );
+
+    auto sourceBuffer = m_originalBuffers.find( filePath );
+    string nodeName = node.name();
+
+    if( sourceBuffer!=m_originalBuffers.end() &&
+        offset > 0 && offset + nodeName.size() < sourceBuffer->second.size() &&
+        sourceBuffer->second.substr( offset, nodeName.size() ) == nodeName )
+    {
+      line = std::count( sourceBuffer->second.begin(), sourceBuffer->second.begin() + offset, '\n' ) + 1;
+      offsetInLine = offset - sourceBuffer->second.rfind( '\n', offset );
+    }
+    else
+    {
+      std::ostringstream oss;
+      oss << "Node from file=" << filePath << ", path=" << node.path() << ", offset=" << offset;
+      oss << " could not be found: ";
+      oss << ( sourceBuffer == m_originalBuffers.end() ?
+               ( filePathAtt.empty() ? "Source file not found." : "Source file info not loaded") :
+               ( offset > 0 && offset + nodeName.size() < sourceBuffer->second.size() ?
+                 "The offset doesn't lead to this node." :
+                 "Offset out of bounds." )
+               );
+      oss << "\nSource files list:";
+      for( auto const & buffer : m_originalBuffers )
+        oss << "\n  " << buffer.first << " buffer size = " << std::to_string( buffer.second.size() );
+
+      throw InputError( oss.str() );
+    }
+  }
+
+  return xmlNodePos( *this, filePath, line, offsetInLine, offset );
+}
+
+xmlNodePos::xmlNodePos( xmlDocument const & document_, string const & filePath_, size_t line_,
+                        size_t offsetInLine_, size_t offset_ ):
+  xmlAttributePos( filePath_, line_, offsetInLine_, offset_ ),
+  document( document_ )
+{}
+
+bool xmlNodePos::isFound() const
+{ return line != xmlDocument::npos; }
+
+size_t findTagEnd( string const & xmlBuffer, size_t const offset )
+{
+  bool outOfQuotes = true;
+  auto bufferEnd = xmlBuffer.cend();
+  for( auto it = xmlBuffer.cbegin() + offset; it != bufferEnd; ++it )
+  {
+    if( *it == '"' && *(it - 1) != '\\' )
+    {
+      outOfQuotes = !outOfQuotes;
+    }
+    else if( outOfQuotes && *it == '>' )
+    {
+      return it - xmlBuffer.cbegin();
+    }
+  }
+  return string::npos;
+}
+size_t findAttribute( string const & attName, string const & xmlBuffer, size_t const tagBegin, size_t const tagEnd )
+{
+  if( !attName.empty())
+  {
+    size_t searchStart = tagBegin;
+    try
+    {
+      std::smatch m;
+      // we search for a string which is the attribute name followed by an '=', eventually separated by spaces
+      if( std::regex_search( xmlBuffer.cbegin() + searchStart, xmlBuffer.cbegin() + tagEnd,
+                             m, std::regex( attName + "\\s*=\\s*\"" )))
```

Comment:
That does not feel good to regex some `xml` string.
What are you trying the do in the end?

Could a `xpath` query solve your problem?

Reply:
I want to determine the attribute line here.

As mentioned in [previous messages](https://github.com/GEOS-DEV/GEOS/pull/2404/#discussion_r1253223108), this portion of code isn't perfect, but takes into account the typical GEOS XMLs. And if the attribute position cannot be determined, the parent node line will be displayed (or in less likely cases a near position of the attribute).

`pugixml` xpaths can only get node and attribute objects. And unfortunately, `pugi::xpath_query` only expose a character offset when an erroneous xpath is evaluated (and the `PUGIXML_NO_EXCEPTIONS` must be enabled).
The code proposed here already get a reliable node character offset from `pugixml`, but `pugi::attribute` don't expose this offset value, that's why I rely on a regex to get the attribute line.

Reply:
OK, I get it better now. I think that `xpath v2` supports `regex`, but `pugixml` only support `v1` 🤷 !
Can you put a little bit more comments around this line? Since it's unusual, I think it's worth documenting the process.

Reply:
OK, I'll add some comments here.